src/Symfony/Component/Notifier/Bridge/FakeSms/FakeSmsTransport.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\FakeSms;
+
+use Symfony\Component\Mailer\Exception\TransportExceptionInterface;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Mime\Email;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+
+/**
+ * @author James Hemery <james@yieldstudio.fr>
+ */
+final class FakeSmsTransport extends AbstractTransport
+{
+    protected const HOST = 'email';
+
+    private $to;
+    private $from;
+    private $mailer;
+
+    public function __construct(string $to, string $from, MailerInterface $mailer = null)
```

Comment:
Please move mailer to the beginning and add the missing parent constructor arguments (httpclient, eventdispatcher).

Plus please Check how the mercure notifier objects the service and do the same with the mailer service id.

Please also check the Mercure notifier DSN for inspiration.

Mailer should also be required, otherwise the transport would not work